datasets/flwr_datasets/partitioner/distribution_partitioner_test.py

```diff
@@ -0,0 +1,211 @@
+"""Test cases for DistributionPartitioner."""
+
+import unittest
+from collections import Counter
+from typing import Any, Dict, List, Tuple, Union
+
+import numpy as np
+from parameterized import parameterized
+
+from datasets import Dataset
+from flwr_datasets.common.typing import NDArrayFloat, NDArrayInt
+from flwr_datasets.partitioner.distribution_partitioner import DistributionPartitioner
+
+
+def _dummy_dataset_setup(
+    num_samples: int, partition_by: str, num_unique_classes: int
+) -> Dataset:
+    """Create a dummy dataset for testing."""
+    data = {
+        partition_by: np.tile(
+            np.arange(num_unique_classes), num_samples // num_unique_classes + 1
+        )[:num_samples],
+        "features": np.random.randn(num_samples),
+    }
+    return Dataset.from_dict(data)
+
+
+def _dummy_distribution_setup(
+    num_partitions: int,
+    num_unique_labels_per_partition: int,
+    num_unique_labels: int,
+    random_mode: bool = False,
+) -> Union[NDArrayFloat, NDArrayInt]:
+    """Create a dummy distribution for testing."""
+    num_columns = num_unique_labels_per_partition * num_partitions / num_unique_labels
+    if random_mode:
+        rng = np.random.default_rng(2024)
+        return rng.integers(1, 10, size=(num_unique_labels, int(num_columns)))
+    return np.tile(np.arange(num_columns) + 1.0, (num_unique_labels, 1))
+
+
+# pylint: disable=R0913
+def _get_partitioner(
+    num_partitions: int,
+    num_unique_labels_per_partition: int,
+    num_samples: int,
+    num_unique_labels: int,
+    preassigned_num_samples_per_label: int,
+    rescale_mode: bool = True,
+) -> Tuple[DistributionPartitioner, Dict[int, Dataset]]:
+    """Create DistributionPartitioner instance."""
+    dataset = _dummy_dataset_setup(
+        num_samples,
+        "labels",
+        num_unique_labels,
+    )
+    distribution = _dummy_distribution_setup(
+        num_partitions,
+        num_unique_labels_per_partition,
+        num_unique_labels,
+    )
+    partitioner = DistributionPartitioner(
+        distribution_array=distribution,
+        num_partitions=num_partitions,
+        num_unique_labels_per_partition=num_unique_labels_per_partition,
+        partition_by="labels",
+        preassigned_num_samples_per_label=preassigned_num_samples_per_label,
+        rescale=rescale_mode,
+    )
+    partitioner.dataset = dataset
+    partitions: Dict[int, Dataset] = {
+        pid: partitioner.load_partition(pid) for pid in range(num_partitions)
+    }
+
+    return partitioner, partitions
+
+
+class TestDistributionPartitioner(unittest.TestCase):
+    """Unit tests for DistributionPartitioner."""
+
+    @parameterized.expand(  # type: ignore
+        [
+            # num_partitions, num_unique_labels_per_partition, num_samples,
+            # num_unique_labels, preassigned_num_samples_per_label
+            (10, 2, 200, 10, 5),
+            (10, 2, 200, 10, 0),
+            (20, 1, 200, 10, 5),
```

Comment:
If all the methods use the same parameters we can use `parameterized_class` instead of ` @parameterized.expand(`. See https://github.com/adap/flower/blob/main/datasets/flwr_datasets/federated_dataset_test.py